### PR TITLE
Metafields tutorial javascript

### PR DESCRIPTION
**Pull request by jrrafols**
A JS version of the customer account extension and writing to metafields tutorial. 

This is a copy of the [React version of the tutorial code](https://github.com/Shopify/customer-account-tutorials/tree/main/react/example-customer-account--metafields--react), but with updates to:
* [shopify.extension.toml](https://github.com/Shopify/customer-account-tutorials/compare/metafields-tutorial-js?expand=1#diff-7773119742b4669e65a424da421d03f684428a780a4c23d25d0d51a4fcd45598)
* [ProfilePreferenceExtension.js](https://github.com/Shopify/customer-account-tutorials/compare/metafields-tutorial-js?expand=1#diff-8aa44f9b4bb419ea56e1b06130f66ca6a6ac0670444cefc0baebbb72c1ce55a3)

The Remix part of the tutorial remains in React & TS, but the extension itself is now Vanilla JS.

Contact me for testing steps.

### CODE REVIEW

**Comment by lihaokx** [comment]
How can we see how it looks like on public doc?

**Review by lihaokx — APPROVED** [review]
🎩  LGTM

File: javascript/example-customer-account--metafields--js/extensions/customer-preferences/src/ProfilePreferenceExtension.js [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+  const { i18n, ui } = api;
+
+  const setNickName = (value) => {
+    nickName = value;
+  };
+
+  const handleSubmit = async () => {
+    await setCustomerPreferences(customerId, nickName);
+
+    nickNameText.children[0].updateText(nickName);
```
Comment (robin-drexler) [inline javascript/example-customer-account--metafields--js/extensions/customer-preferences/src/ProfilePreferenceExtension.js:37]: Can we put whatever is in `children[0]` into a const and access it this way?

Comment (jrrafols) [reply javascript/example-customer-account--metafields--js/extensions/customer-preferences/src/ProfilePreferenceExtension.js]: Updated to use `createText()` which allows invoking `update` directly on the component

**Pull request merged**